libkbfs/folder_block_ops.go

```diff
@@ -2585,14 +2575,36 @@ func (fbo *folderBlockOps) getDeferredWriteCountForTest(lState *lockState) int {
 	return len(fbo.deferredWrites)
 }
 
+func (fbo *folderBlockOps) updatePointer(kmd KeyMetadata, oldPtr BlockPointer, newPtr BlockPointer) {
+	updated := fbo.nodeCache.UpdatePointer(oldPtr.Ref(), newPtr)
+	if !updated {
+		return
+	}
+	// Prefetch the new reference, but only if it already exists in the block
```

Comment:
We won't suddenly lose our reference if it's been returned already, allowing us to call `NewEmpty` on the existing block.

Maybe we need a better way to initialize typed blocks given the encoded one.